src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -2255,10 +2255,111 @@ public static FormulaValue Error(IRContext irContext, FormulaValue[] args)
                         return CommonErrors.RuntimeTypeMismatch(irContext);
                 }
 
-                result.Add(new ExpressionError { Kind = errorKind, Message = messageField?.Value as string });
+                var message = messageField != null ? messageField.Value : GetDefaultErrorMessage(errorKind);
+                result.Add(new ExpressionError { Kind = errorKind, Message = message });
             }
 
             return result;
+        }
+
+        private static string GetDefaultErrorMessage(ErrorKind errorKind)
+        {
+            switch (errorKind)
+            {
+                case ErrorKind.None:
+                    // Default message that is shown to users when an error was produced, but with kind = 'none'
+                    return "System no error";

```

Comment:
>"System no error" [](http://example.com/codeflow?start=27&length=17)

Do we need to localize those messages? Maybe we could populate MessageKey field so they can get translated automatically.

Reply:
Good point, we definitely should... but I've seen many other messages from the interpreter hardcoded. Let me check with Mike to see what's our story for the interpreter.

Reply:
Messages in the interpreter are not localized today, we have some work items to fix that. While that assembly doesn't have its own resources file, the messages need to stay hardcoded for now.

Reply:
Yes, tracked with #548